pkg-manager/headless/src/index.ts

```diff
@@ -545,7 +546,7 @@ export async function headlessInstall (opts: HeadlessOptions) {
       storeDir: opts.storeDir,
       virtualStoreDir,
     })
-    if (opts.useLockfile) {
+    if (opts.useLockfile && JSON.stringify(wantedLockfile) !== JSON.stringify(opts.wantedLockfileUnmodified)) {
```

Comment:
nit: It might be worth following up with a `deepEqual(...)` function that early exits as soon as it sees the lockfiles are different for performance. I've seen `pnpm-lock.yaml` files grow to be 4MB, and serializing 2 * 4MB could add a few seconds to each `pnpm install`.

Reply:
I have changed it. There was actually no need in the copy.

Reply:
Nice!